lib/loc/diff.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loc
+
+import (
+	"sort"
+
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// RepositoryDiff represents the difference in Docker images for a particular repository.
+type RepositoryDiff struct {
+	// Repository is the Docker image repository.
+	Repository string
+	// Tags are the tags for this repository.
+	Tags []TagDiff
```

Comment:
Have you considered left/right groups of tags instead?
```go
type RepositoryDiff struct {
	// Repository is the Docker image repository.
	Repository string
	// Left/Right define the differences between repositories
	Left, Right []string
}
```

Reply:
The current data structure is more convenient to operate with on the caller side.